src/pages/_app.tsx

```diff
@@ -42,6 +45,14 @@ const AccountApp: React.FC = ({ children }) => {
                     "An error occurred while fetching the data.",
                   );
 
+                  if (
+                    response.status === HttpStatus.UNAUTHORIZED &&
+                    response.statusText === "Unauthorized"
+                  ) {
+                    router && router.replace(router.pathname);
+                    return;
+                  }
+
```

Comment:
Here, I catch the unauthorized web error, and trigger a page reload to retrigger all auth flow. This piece is here because of the revalidate on focus feature from SWR, where when you come back on a left opened tab, the api calls are retriggered to update data. But, as the auth middleware ignores the api calls to do refresh flow, we need to reload the page to make everything work smoothly.